Sources/_StringProcessing/Regex/DSLTree.swift

```diff
@@ -455,3 +452,225 @@ public struct CaptureTransform: Hashable, CustomStringConvertible {
     "<transform result_type=\(resultType)>"
   }
 }
+
+// MARK: AST wrapper types
+//
+// These wrapper types are required because even @_spi-marked public APIs can't
+// include symbols from implementation-only dependencies.
+
+extension DSLTree {
+  /// Presents a wrapped version of `DSLTree.Node` that can provide an internal
+  /// `_TreeNode` conformance.
+  struct _Tree: _TreeNode {
```

Comment:
What is this used for? Why isn't this `DSLTree`?

Reply:
This is more or less a compiler limitation workaround. `_TreeNode` is a protocol defined in `_RegexParser`, and since `DSLTree` is public (as SPI), its conformance to `_TreeNode` is also public. However, `_RegexParser` is imported as implementation only, so its symbols can't/shouldn't be included in the public interface of `_StringProcessing`. So this type basically provides that `DSLTree: _TreeNode` conformance by composition, and since it's internal, it doesn't cause the issue with exported implementation-only symbols.